External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp

```diff
@@ -1544,7 +1544,7 @@ void OpDispatchBuilder::SAHFOp(OpcodeArgs) {
   OrderedNode *Src = LoadGPRRegister(X86State::REG_RAX, 1, 8);
 
   // Clear bits that aren't supposed to be set
-  Src = _And(Src, _Constant(~0b101000));
+  Src = _Andn(Src, _Constant(0b101000));
```

Comment:
Is there a good reason why the optimizer doesn't do this one? Would it help in cases where the guest code does masking itself?

Obviously this makes sense in addition, for compile-time reasons, but for the code reduction it smells funny

Reply:
This line actually doesn't do any code reduction by itself, the real optimization is in the `SetPackedRFLAG` code. This one is more about signaling intent that we want to clear some bits. ConstProp pass should theoretically in the future see if the immediate fits in the encoding space and does a `mov + bic` or `and <reg>, #<negated constant>`. This isn't the case today sadly.

Reply:
Ah, I see. If it's just intent then, makes sense to me :+1: Thanks for explaining